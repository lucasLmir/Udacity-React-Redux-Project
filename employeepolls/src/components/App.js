import { Route, Routes } from "react-router-dom";
import { Fragment, useEffect } from "react";
import Nav from "./Nav";
import Dashboard from "./Dashboard";
import { connect } from "react-redux";
import { handleInitialData } from "../actions/shared";
import LoadingBar from "react-redux-loading-bar";
import NewQeustion from "./NewQuestion";
import QuestionPage from "./QuestionPage";
import LoginPage from "./LoginPage";
import LeaderBoard from "./LeaderBoard";
import NotFound from "./NotFound";
import ProtectedRoutes from "./ProtectedRoutes";

function App(props) {
  useEffect(() => {
    props.dispatch(handleInitialData());
  }, []);

  return (
    <Fragment>
      <LoadingBar />
      {props.loading === true ? null : (
        <div className="container">
          <Nav />
          <Routes>
            <Route element={<ProtectedRoutes />}>
              <Route path="/" exact element={<Dashboard />} />
              <Route path="/add" element={<NewQeustion />} />
              <Route path="/question/:id" element={<QuestionPage />} />
              <Route path="/leaderboard" element={<LeaderBoard />} />
              <Route path="*" element={<NotFound />} />
            </Route>
            <Route path="/login" element={<LoginPage />} />
          </Routes>
        </div>
      )}
    </Fragment>
  );
}

const mapStateToProps = ({ users }) => ({
  loading: users === null,
});

export default connect(mapStateToProps)(App);
