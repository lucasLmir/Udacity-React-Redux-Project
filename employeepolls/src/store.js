import reducer from "./reducers";
import { createStore } from "redux";
import middleware from "./middleware";

export const store = createStore(reducer, middleware);