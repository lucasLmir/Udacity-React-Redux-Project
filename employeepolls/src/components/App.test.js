import { render } from "@testing-library/react";
import App from "./App";
import { store } from "../store";
import "@testing-library/jest-dom";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import { setAuthedUser } from "../actions/authedUser";

describe("App", () => {
  it("is expected to render the component", () => {
    const component = render(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
  it("It is expected to load the Login Page if no authed User is logged.", () => {
    const component = render(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    );
    expect(component.getByTestId("login-form")).toBeInTheDocument();
  });

  it("It is expected to give access to the app if a user is logged in.", async () => {
    const user = "mtsamis";
    await store.dispatch(setAuthedUser(user));
    const component = render(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    );
    expect(component.queryByTestId("login-form")).not.toBeInTheDocument();
  });
});
