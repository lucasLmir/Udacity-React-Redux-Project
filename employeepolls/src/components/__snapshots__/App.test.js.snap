// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App is expected to render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="direction: ltr;"
      >
        <div
          class=""
          style="width: 0%; transition: width 800ms linear 0s; will-change: width, opacity; height: 3px; background-color: red; position: absolute; opacity: 0;"
        />
        <div
          style="display: table; clear: both;"
        />
      </div>
      <div
        class="container"
      >
        <nav
          class="nav"
        >
          <ul>
            <li>
              <a
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                href="/new"
              >
                Ask a Question
              </a>
            </li>
            <li>
              <a
                href="/leaderboard"
              >
                Leader Board
              </a>
            </li>
            <li
              class="right"
            >
              <a
                href="/login"
              >
                <span
                  data-testid="userName"
                >
                  Login
                </span>
              </a>
            </li>
            <li>
              <a
                href="/login"
              >
                <img
                  alt="avatar of undefined"
                  class="small-avatar"
                  data-testid="avatarImg"
                  hidden=""
                />
              </a>
            </li>
          </ul>
        </nav>
        <div>
          <button
            data-testid="logout-btn"
            hidden=""
          >
            LogOut
          </button>
          <form
            data-testid="login-form"
          >
            <input
              data-testid="uname-input"
              id="Uname"
              name="Uname"
              placeholder="Username"
              type="text"
              value=""
            />
            <br />
            <input
              data-testid="pass-input"
              id="Pass"
              name="Pass"
              placeholder="Password"
              type="Password"
              value=""
            />
            <br />
            <button
              data-testid="submit-btn"
            >
              LogIn
            </button>
          </form>
          <p
            data-testid="message"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="direction: ltr;"
    >
      <div
        class=""
        style="width: 0%; transition: width 800ms linear 0s; will-change: width, opacity; height: 3px; background-color: red; position: absolute; opacity: 0;"
      />
      <div
        style="display: table; clear: both;"
      />
    </div>
    <div
      class="container"
    >
      <nav
        class="nav"
      >
        <ul>
          <li>
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/new"
            >
              Ask a Question
            </a>
          </li>
          <li>
            <a
              href="/leaderboard"
            >
              Leader Board
            </a>
          </li>
          <li
            class="right"
          >
            <a
              href="/login"
            >
              <span
                data-testid="userName"
              >
                Login
              </span>
            </a>
          </li>
          <li>
            <a
              href="/login"
            >
              <img
                alt="avatar of undefined"
                class="small-avatar"
                data-testid="avatarImg"
                hidden=""
              />
            </a>
          </li>
        </ul>
      </nav>
      <div>
        <button
          data-testid="logout-btn"
          hidden=""
        >
          LogOut
        </button>
        <form
          data-testid="login-form"
        >
          <input
            data-testid="uname-input"
            id="Uname"
            name="Uname"
            placeholder="Username"
            type="text"
            value=""
          />
          <br />
          <input
            data-testid="pass-input"
            id="Pass"
            name="Pass"
            placeholder="Password"
            type="Password"
            value=""
          />
          <br />
          <button
            data-testid="submit-btn"
          >
            LogIn
          </button>
        </form>
        <p
          data-testid="message"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
